docs(graphql): document fieldResolverEnhancers (#1144)
nestjs/docs.nestjs.com: content/graphql/guards-interceptors.md

```

> info **Hint** In the above example, we have assumed that the `user` object is assigned to the context of your GraphQL application.

> #### Execute enhancers at the field/method level

Reviewer: `/method` sounds confusing here. Query/mutation handlers are methods as well

> #### Interceptors at the `@ResoleField()` method level

Reviewer 2: ```suggestion
#### Interceptors at the `@ResolveField()` method level
```

> #### Interceptors at the `@ResolveField()` method level

Reviewer 2: ```suggestion
#### Execute enhancers at the field/method level
```

> In a GraphQL context, interceptors [do not access the full GraphQL response](https://github.com/nestjs/graphql/issues/320#issuecomment-511193229): they only get the response of the `@Query()`/`@Mutation()` method, but not the complete response with the resolved fields. You can tell Nest to execute interceptors (as well as guards and filters) for methods annotated `@ResolveField()` by setting the `fieldResolverEnhancers` option in `GqlModuleOptions`:

Reviewer 2: ```suggestion
In the GraphQL context, Nest does not run **enhancers** (the generic name for interceptors, guards and filters) at the field level [see this issue](https://github.com/nestjs/graphql/issues/320#issuecomment-511193229): they only run for the top level `@Query()`/`@Mutation()` method. You can tell Nest to execute interceptors, guards or filters for methods annotated with `@ResolveField()` by setting the `fieldResolverEnhancers` option in `GqlModuleOptions`.  Pass it a list of `'interceptors'`, `'guards'`, and/or `'filters'` as appropriate:
```

```typescript
GraphQLModule.forRoot({
  fieldResolverEnhancers: ['interceptors']
}),
```

> > **Warning** Enabling interceptors for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of interceptors that are not strictly necessary for your field resolvers. You can do this using the following helper method:

Reviewer 2: ```suggestion
> **Warning** Enabling interceptors for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of interceptors that are not strictly necessary for your field resolvers. You can do this using the following helper function:
```

> > **Warning** Enabling interceptors for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of interceptors that are not strictly necessary for your field resolvers. You can do this using the following helper function:

Reviewer 2: ```suggestion
> **Warning** Enabling enhancers for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of enhancers that are not strictly necessary for your field resolvers. You can do this using the following helper function:
```

```typescript
export function isResolvingGraphQLField(context: ExecutionContext): boolean {

>   if (context.getType().toString() === 'graphql') {

Reviewer: One little suggestion: you can use the `GqlContextType` as stated here https://docs.nestjs.com/fundamentals/execution-context#current-application-context

```typescript
if (context.getType<GqlContextType>() === 'graphql') {
```

}),
```

> **Warning** Enabling interceptors for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of interceptors that are not strictly necessary for your field resolvers. You can do this using the following helper function:

```typescript
export function isResolvingGraphQLField(context: ExecutionContext): boolean {

>   if (context.getType<GqlContextType>().toString() === 'graphql') {

Reviewer 2: I think you can simplify further (see http://localhost:4200/fundamentals/execution-context#current-application-context)

```suggestion
  if (context.getType<GqlContextType>() === 'graphql') {
```

Author: Had missed that from @kamilmysliwiec's comment.
